File: bubble_sort/javascript/bubble_sort.js
class BubbleSort {
    sort(input) {
        // Make as many passes through the input as there are items.
        for (let passIndex = 0; passIndex < input.length; passIndex++) {
            for (let index = 0; index < input.length; index++) {
                let firstItem = input[index];
                let secondItem = input[index + 1]
    
                if (firstItem > secondItem) {
                    input[index] = secondItem;
                    input[index + 1] = firstItem;
                }
            }
        }

        return input;
    }
}

bubbleSort = new BubbleSort();
let sorted = bubbleSort.sort([9,6,4,5,8,2]);
console.log(sorted);

File: bubble_sort/c/bubble_sort.c
int *sort(int *input, int size)
{
    // Make as many passes through the input as there are items.
    for (int passIndex = 0; passIndex < size; passIndex++)
    {
        // Now iterate the input and do the swapping.
        for (int index = 0; index < size; index++)
        {
            int firstItem = input[index];
            int secondItem = input[index + 1];

            if (firstItem > secondItem)
            {
                input[index] = secondItem;
                input[index + 1] = firstItem;
            }
        }
    }

    return input;
}
